### ISSUE TITLE
How To?

### ISSUE THREAD

**Issue by Balderick** [OSVR/OSVR-Android-Samples#10]
Have a working build of osvr-android-build in android studio but build fails in vs 2017 community.

Have cmake configured with osvr-andoid environment variable for osvr-android-build but have no ideo what i am supposed to do with osvr-android-samples or what to do with osvr-android build because of 
> Build instructions
>
>    Follow the instructions for building OSVR-Android-Build here: https://github.com/OSVR/OSVR-Android-Build You do not need to install the osvr_server binaries to the device, as these samples currently use the joint client kit functionality.

Could it be possible for anybody to share how they used android studio to get working osvr-android-samples working on on an android device and expand on what is described in https://github.com/OSVR/OSVR-Android-Samples/blob/master/README.md ?

All i want to do is https://www.youtube.com/watch?v=KdoLiMAA2lk

**Comment by DuFF14** [comment]
@Balderick Still a WIP, but the androidPalace branch of OSVR-Unity-Palace-Demo (Unity 5.3.5) contains the libraries from the latest OSVR-Android-Build: https://github.com/OSVR/OSVR-Unity-Palace-Demo/tree/androidPalace

First copy the following osvr_server_config.json to /sdcard/osvr/
```json
/* The default config file, as used with the HDK, is fully auto-detected, so no configuration required. */
/* See the JSON-Schema for server config or other samples to see possible options. */
{
  "display": {
    "meta": {
      "schemaVersion": 1
    },
    "hmd": {
      "device": {
        "vendor": "Samsung",
        "model": "Galaxy S6",
        "num_displays": 1,
        "Version": "1.1",
        "Note": "Samsung Galaxy S6"
      },
      "field_of_view": {
        "monocular_horizontal": 90,
        "monocular_vertical": 101.25,
        "overlap_percent": 100,
        "pitch_tilt": 0
      },
      "resolutions": [
        {
          "width": 2560,
          "height": 1440,
          "video_inputs": 1,
          "display_mode": "horz_side_by_side",
          "swap_eyes": 0
        }
      ],
       "distortion": {
      "k1_red": 0.5,
      "k1_green": 0.5,
      "k1_blue": 0.5
    },
      "rendering": {
        "right_roll": 0,
        "left_roll": 0
      },
      "eyes": [
        {
          "center_proj_x": 0.5,
          "center_proj_y": 0.5,
          "rotate_180": 0
        },
        {
          "center_proj_x": 0.5,
          "center_proj_y": 0.5,
          "rotate_180": 0
        }
      ]
    }
  },
  "plugins": [
    "com_osvr_android_sensorTracker"
  ]
}
```
It may auto-generate the file if you forget that step.

Here is a link to APK which works on my Galaxy S6:
https://drive.google.com/open?id=0B0X-T5b2-zcOR2c4UVF1SWxsS1U

Latency is high because prediction is off. I still need to add Android support to the OSVR-Unity-Rendering plugin to connect it to RM.

**Comment by DuFF14** [comment]
you also should modify the "width" and "height" properties in the config to match your phone's display.

**Comment by Balderick** [comment]
;-)   I was looking at osvr-cardboard earlier which uses google cardboard qr info to generate an osvr display descriptor for your device.

**Comment by DuFF14** [comment]
There is an AndroidPlugin.jar in the Unity plugin for which the source is not yet included, which is responsible for loading the server plugins from C#: https://github.com/OSVR/OSVR-Unity-Palace-Demo/blob/androidPalace/Assets/OSVRUnity/src/ClientKit.cs#L103

Here is a screenshot of my Android Studio project for that library:
![image](https://cloud.githubusercontent.com/assets/10972018/26122474/a3bc7d46-3a45-11e7-862a-fc2a8c2aefe7.png)

AndroidPlugin.jar will be replaced by a newer Android library in future builds.
This link was helpful: http://www.thegamecontriver.com/2015/04/android-plugin-unity-android-studio.html

**Comment by Balderick** [comment]
I found using the pre compiled apk for the unity android branch of palace demo to not be using rotational head tracking.  The right stick on shield controller panned camera view left or right only.  No up or down.
The shield tablet was using full screen rendering and not SBS unlike googlevr apps.

Tried onnecting osvr hdk to shield tablet with HDMI and was surprised both console and mirror mode rendered android environment in hdk.  Palace demo still did not give head tracking (was expecting working rotational) and HDK was not using appropriate distortion and other display settings so felt like I was really squint eyed with eyes on side of head with a blind spot in middle of fov.

No notifications were shown when starting osvr and exiting osvr server.

No OSVR directory exists on my shield tablet SD card after installing the palace demo apk from android branch 
Will keep on trying to compile from source and experimenting/playing
